From: Takashi Iwai <tiwai@suse.de>
Subject: Fix kABI breakage due to acpi_ec gpe field change
Patch-mainline: Never, SLE15 kABI fix
References: bsc#1051510

patches.drivers/ACPI-EC-Fix-debugfs_create_-usage changed acpi_ec.gpe
field type, and it broke kABI.  Add an ugly ifdef as usual.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 drivers/acpi/internal.h |    9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

--- a/drivers/acpi/internal.h
+++ b/drivers/acpi/internal.h
@@ -159,7 +159,14 @@ static inline void acpi_early_processor_
    -------------------------------------------------------------------------- */
 struct acpi_ec {
 	acpi_handle handle;
-	u32 gpe;
+#ifdef __GENKSYMS__
+	unsigned long gpe;
+#else
+	union {
+		u32 gpe;
+		unsigned long __orig_gpe;
+	};
+#endif
 	unsigned long command_addr;
 	unsigned long data_addr;
 	bool global_lock;
